src/main/java/de/tum/in/www1/artemis/config/LiquibaseConfiguration.java

```diff
@@ -64,4 +89,52 @@ public SpringLiquibase liquibase(@LiquibaseDataSource ObjectProvider<DataSource>
         }
         return liquibase;
     }
+
+    private String getPreviousVersionElseThrow() {
+        try (var statement = createStatement()) {
+            statement.executeQuery("SELECT * FROM DATABASECHANGELOG;");
+            var result = statement.executeQuery("SELECT latest_version FROM artemis_version;");
+            statement.closeOnCompletion();
+            if (result.next()) {
+                return result.getString("latest_version");
+            }
+            return null;
+        }
+        catch (SQLException e) {
+            log.info(e.getMessage());
+            // if no changelog or no version cane be found, it means it is not yet available, we do not need to throw an exception
+            return null;
+        }
+    }
+
+    private Statement createStatement() throws SQLException {
+        var connection = dataSource.getConnection();
+        return connection.createStatement();
+    }
+
+    /**
+     * Stores the current version in the database after the application is ready
+     * @param event used to prevent this method for running in tests
+     */
+    @EventListener()
+    public void storeCurrentVersionToDatabase(ApplicationReadyEvent event) {
+        if (event.getApplicationContext().getEnvironment().acceptsProfiles(Profiles.of(SPRING_PROFILE_TEST))) {
```

Comment:
Same as line 63; Can you move this into a method?

Reply:
done